p2panda-rs/src/document/document.rs

```diff
@@ -265,32 +270,79 @@ impl DocumentBuilder {
     }
 }
 
+impl<T: AsOperation + WithOperationID + WithPublicKey> From<Vec<&T>> for DocumentBuilder {
+    fn from(operations: Vec<&T>) -> Self {
+        let operations = operations
```

Comment:
Not very DRY in these conversion traits...

Reply:
Yeah! I think it's alright, if you want you can use the `where` syntax if that makes it more readable?

I hope that Rust introduces a feature soon where one can define trait requirements similarily like `type`.